cpp/src/parquet/column_reader.h

```diff
@@ -368,6 +368,11 @@ class PARQUET_EXPORT RecordReader {
 
   virtual void DebugPrintState() = 0;
 
+  /// \brief Returns the dictionary owned by the current decoder. Throws an
+  /// exception if the current decoder is not for dictionary encoding.
+  /// \param[out] dictionary_length The number of dictionary entries.
+  virtual const void* ReadDictionary(int32_t* dictionary_length) = 0;
```

Comment:
I think it should be possible to make this templated (with implementations in the .cc file?  If so I think that might make sense for consumers.  Either way we should clarify how users can understand what to case the dictionary to.

Reply:
Added some explanations and examples in the function comments. PTAL